src/Common/src/System/Net/Sockets/SocketAPMExtensions.cs

```diff
@@ -829,6 +829,24 @@ public static int EndSendTo(this Socket socket, IAsyncResult asyncResult)
         {
             return TaskToApm.End<int>(asyncResult);
         }
+
+        // Behavior adapter.
+        private static ArraySegment<byte> CreateArraySegment(byte[] buffer, int offset, int size)
+        {
+            if (buffer == null)
+            {
+                throw new ArgumentNullException("buffer");
+            }
+
+            try
+            {
+                return new ArraySegment<byte>(buffer, offset, size);
+            }
+            catch (ArgumentException ae)
```

Comment:
ArgumentOutOfRangeException is an ArgumentException, so you could add a catch block before this that just propagates rather than creating a new exception, e.g.
```C#
catch (ArgumentOutOfRangeException)
{
    throw;
}
catch (ArgumentException ae)
{
    ...
}
```